public/js/NeighbourPlot.js

```diff
@@ -0,0 +1,199 @@
+var d3 = require('d3');
+require('d3-tip')(d3);
+
+/**
+ * Draws PCA scatterplot of the data Clicking on scatterplot points will
```

Comment:
I'd bury the different dimensionality reduction options deep in an "advanced" menu somewhere... I think we can do pretty well ourselves figuring out how we want to show the data, and it's good not to overwhelm newbies with too many options!

As to the docstring, I specifically tried to design something that would not be too egregious in violating JSDoc syntax. I think the above would render fine *even if* we eventually compile the JSDoc to HTML; it would just be a bit repetitive.